x-pack/plugin/repository-encrypted/src/main/java/org/elasticsearch/repositories/encrypted/DecryptionPacketsInputStream.java

```diff
@@ -124,19 +123,19 @@ public void reset() throws IOException {
         throw new IOException("Mark/reset not supported");
     }
 
+    @Override
+    public void close() throws IOException {
+        IOUtils.close(super::close, source);
+    }
+
     private int decrypt(PrefixInputStream packetInputStream) throws IOException {
         // read only the IV prefix into the packet buffer
         int ivLength = packetInputStream.readNBytes(packetBuffer, 0, GCM_IV_LENGTH_IN_BYTES);
         if (ivLength != GCM_IV_LENGTH_IN_BYTES) {
             throw new IOException("Packet heading IV error. Unexpected length [" + ivLength + "].");
         }
-        // extract the nonce and the counter from the packet IV
-        ByteBuffer ivBuffer = ByteBuffer.wrap(packetBuffer, 0, GCM_IV_LENGTH_IN_BYTES).order(ByteOrder.LITTLE_ENDIAN);
-        int packetIvNonce = ivBuffer.getInt(0);
-        long packetIvCounter = ivBuffer.getLong(Integer.BYTES);
-        if (packetIvNonce != nonce) {
-            throw new IOException("Packet nonce mismatch. Expecting [" + nonce + "], but got [" + packetIvNonce + "].");
-        }
+        // extract the counter from the packet IV
+        long packetIvCounter = ByteUtils.readLongLE(packetBuffer, Integer.BYTES);
```

Comment:
Fair point, it now looks like this:
```
        // extract the counter from the packet IV and validate it (that the packet is in order)
        // skips the first 4 bytes in the packet IV, which contain the encryption nonce, which cannot be explicitly validated
        // because the nonce is not passed in during decryption, but it is implicitly because it is part of the IV,
        // when GCM validates the packet authn tag
        long packetIvCounter = ByteUtils.readLongLE(packetBuffer, Integer.BYTES);
        if (packetIvCounter != counter) {
            throw new IOException("Packet counter mismatch. Expecting [" + counter + "], but got [" + packetIvCounter + "].");
        }
```